### PR TITLE
504 authentication support for faststream

### LINKED ISSUE #504: Authentication support for FastStream
- [x] write chapter in design doc with proposed API
- [x] agree with everyone on the API
- [x] write implementation for Kafka
- [x] raise NotImplemented for other brokers
- [x] write tasks for other brokers

### PR DESCRIPTION
**Pull request by Sternakt**
Closes #504

### CODE REVIEW

File: faststream/kafka/broker.py [hunk 1]
```diff
@@ -368,6 +378,7 @@ def subscriber(  # type: ignore[override]
             max_poll_records=max_poll_records,
             exclude_internal_topics=exclude_internal_topics,
             isolation_level=isolation_level,
+            **parse_security(self.security),
```
Comment (Lancetnik) [inline faststream/kafka/broker.py:381]: You need to remove these (security) arguments (and add your security) from
* kafka/broker.pyi __init__, connect
* kafka/shared/schemas.py
* kafka/fastapi.pyi __init__

**Review by Lancetnik — CHANGES_REQUESTED** [review]
To merge this PR, please fix annotations and add 2 a few tests to `tests/asyncapi/kafka/test_connection.py` and `tests/asyncapi/kafka/test_fastapi.py` to validated generated AsyncAPI spec

File: faststream/kafka/broker.py [hunk 2]
```diff
@@ -93,6 +94,13 @@ def __init__(
             client_id (str): The client ID for the Kafka client.
             **kwargs: Additional keyword arguments.
         """
+        if protocol is None:
+            security = kwargs.get("security")
```
Comment (Lancetnik) [inline faststream/kafka/broker.py:98]: Why are you getting it from kwargs? Add it to arguments directly, it is much explicit way (but don't forget to pass it in super)

File: faststream/kafka/broker.py (hunk 1 above)
Comment (Lancetnik) [reply faststream/kafka/broker.py:381]: Also, remove these line from here: Producer and Consumer shared arguments are passing to Consumer connection in the 179 and 199 lines

File: tests/brokers/kafka/test_security.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        pass
+
+
+@pytest.mark.asyncio
+async def test_base_security():
+    with patch_aio_consumer_and_producer() as (consumer, producer):
+        from docs.docs_src.kafka.basic_security.app import broker as basic_broker
+
+        async with basic_broker:
+            await basic_broker.start()
```
[the remark this answers is not in the document]
Comment (Lancetnik) [reply tests/brokers/kafka/test_security.py:28]: By the way, if it is a tests require real Kafka instance, please mark them by `@pytest.mark.kafka`

File: faststream/kafka/broker.py (hunk 1 above)
Comment (Lancetnik) [reply faststream/kafka/broker.py:381]: Probably, you don't need to patch `kafka/shared/schemas.py`, but unpuck `parse_security` in the `_connect` method in the 179 line

File: faststream/kafka/broker.py (hunk 2 above)
Comment (Sternakt) [reply faststream/kafka/broker.py]: Fixed

File: faststream/rabbit/security.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from typing import Any, Dict, Optional
+
+from faststream.broker.security import BaseSecurity
+
+
+def parse_security(security: Optional[BaseSecurity]) -> Dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (Sternakt) [reply faststream/rabbit/security.py:6]: Used in RabbitBroker when [creating a robust connection](https://github.com/airtai/fastkafka/pull/573/files#diff-0456fbc42e3f97b3d244626c861c18b69d6924bd724f7f86a808589016969fd1R171)

File: faststream/kafka/broker.py (hunk 1 above)
Comment (Sternakt) [reply faststream/kafka/broker.py]: > You need to remove these (security) arguments (and add your security) from
> 
> * kafka/broker.pyi **init**, connect
> * kafka/shared/schemas.py
> * kafka/fastapi.pyi **init**

- Updated signatures, schema left the same 
- Unpacking of security params is moved to _connect method

**Review by davorrunje — APPROVED** [review]

**Pull request merged**